wallet/wallet-controller/src/lib.rs

```diff
@@ -149,16 +164,21 @@ impl<T: NodeInterface> Controller<T> {
     }
 }
 
-pub type RpcController = Controller<NodeRpcClient>;
-pub type HandlesController = Controller<WalletHandlesClient>;
+impl<T> Drop for Controller<T> {
+    fn drop(&mut self) {
+        // TODO: Wait for the sync task to stop
+        self.sync_handle.abort();
+    }
+}
 
 pub async fn make_rpc_controller(
+    chain_config: Arc<ChainConfig>,
```

Comment:
What's the point of passing chain_config and returning it?

Reply:
Sorry, what do you mean? `chain_config` is passed to the `Controller::new` call.